hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -63,7 +64,7 @@
  * @param <T> The type of engine-specific record representation, e.g.,{@code InternalRow}
  *            in Spark and {@code RowData} in Flink.
  */
-public final class HoodieFileGroupReader<T> implements Closeable {
+public final class HoodieFileGroupReader<T> implements Closeable, Iterator {
```

Comment:
Should reader implement Iterator? We can simply keep the list of merged records here and then get the iterator out of list at the call site. If you plan to make it an iterator, then implement `ClosableIterator` and rename as `HoodieFileGroupIterator`

Reply:
Sounds good. Will modify the PR accordingly.

Reply:
Hi @codope, a quick question: when the basefile is huge, keeping the list of merged records will cause any issues or not?

Reply:
we should be using `ExternalSpillableMap` which can spill `HoodieRecord` to disk if necessary. But, as you know there is a tradeoff. So, the memory fraction is controlled by a config (max memory for spillable map in `HoodieCommonConfig`).